### PR TITLE
Add support for managed discord server

### PR DESCRIPTION
**Pull request by MOZGIII**
This PR adds support for "managed" discord servers. It's pretty much the same as the usual servers, but the server is not actually created or deleted, instead a `server_id` is required as input at creation, and the deletion does nothing.

### CODE REVIEW

**Comment by Lucky3028** [comment]
@MOZGIII 
Thank you for your contributions, and I'm sorry for being late!!!
Would you mind rebasing master branch b/c CI for the files has broken and the bug was fixed? (I think I can do it tomorrow.)

**Comment by Lucky3028** [comment]
(Ah..."tomorrow" is JST.)

**Comment by MOZGIII** [comment]
Hey @Lucky3028! I should probably do more testing on this before this can be merged. I kind of hacked it together quickly, and I'm not sure it works well.
I'll be getting to configuring the discord server I made it for soon, so I'll give it a try again.

That said, if you feel like it's good enough - feel free to take it in.

I'll rebase in a bit.

**Comment by MOZGIII** [comment]
Rebase done!

**Comment by Lucky3028** [comment]
Thank you!
I've never heard the "managed discord server", so I'll also want to test this when I'm free.

**Comment by Lucky3028** [comment]
II think I want to fix the md lint errors at the same time.

**Comment by MOZGIII** [comment]
Feel free to take over this brach, I'll pull the changes if there are any when I get to it!

File: discord/resource_discord_server.go [hunk 1]
```diff
@@ -10,6 +10,107 @@ import (
...
+func serverSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"server_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
```
Comment (Lucky3028) [inline discord/resource_discord_server.go:22]:
```suggestion
func baseServerSchema() map[string]*schema.Schema {
	return map[string]*schema.Schema{
```

File: discord/resource_discord_server.go [hunk 2]
```diff
@@ -10,6 +10,107 @@ import (
...
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"owner_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+	}
+}
+
```
Comment (Lucky3028) [inline discord/resource_discord_server.go:113]:
```suggestion

func managedServerSchema() map[string]*schema.Schema {
	res := baseServerSchema()

	res["server_id"] = &schema.Schema{
		Type:     schema.TypeString,
		Required: true,
	}
	res["name"] = &schema.Schema{
		Type:     schema.TypeString,
		Optional: true,
	}

	return res
}

func serverSchema() map[string]*schema.Schema {
	res := baseServerSchema()

	res["server_id"] = &schema.Schema{
		Type:     schema.TypeString,
		Computed: true,
	}
	res["name"] = &schema.Schema{
		Type:     schema.TypeString,
		Required: true,
	}

	return res
}

```

File: discord/resource_discord_server.go [hunk 3]
```diff
@@ -20,104 +121,21 @@ func resourceDiscordServer() *schema.Resource {
...
+	return &schema.Resource{
+		CreateContext: resourceServerManagedCreate,
+		ReadContext:   resourceServerRead,
+		UpdateContext: resourceServerUpdate,
+		DeleteContext: resourceServerManagedDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
 		},
+
+		Schema: serverSchema(),
```
Comment (Lucky3028) [inline discord/resource_discord_server.go:138]:
```suggestion
		Schema: managedServerSchema(),
```

**Review by Lucky3028 — CHANGES_REQUESTED** [review]
I tried this PR, and I found needs to be changed.

**Review by Lucky3028 — APPROVED** [review]

**Comment by Lucky3028** [comment]
@MOZGIII Have you tried yet?

**Comment by MOZGIII** [comment]
I'm about to try it in about an hour :D

**Comment by MOZGIII** [comment]
Works

**Review by Lucky3028 — APPROVED** [review]

**Comment by Lucky3028** [comment]
Thank you and LGTM!

**Pull request merged**